### PR TITLE
feat(core): add renderer factory in render3

### PR DESCRIPTION
**Pull request by marclaval**
@mhevery this PR integrates your comments from the previous.

The only thing I haven't changed is that `RendererFactory3` is still the interface to be provided, as it is the case in Angular today. 
We could go for something like `Renderer3 | RendererFactory3` to save a few bytes and keep backward compatibility, but not sure if it's worth it as it would introduce some duck typing.

### CODE REVIEW

File: packages/core/src/render3/instructions.ts [hunk 1]
```diff
@@ -241,18 +243,42 @@ export function createLNode(
...
+  }
   const hostView = host.data !;
   ngDevMode && assertNotEqual(hostView, null, 'hostView');
-  const oldView = enterView(hostView, host);
+  renderComponentOrTemplate(host, hostView, null, template, context);
+  return host;
+}
+
+export function renderComponentOrTemplate<T>(
+    node: LElement, viewState: ViewState, component: T, template?: ComponentTemplate<T>,
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: Isn't `component` same as `context`? Can we merge them?

**Review by mhevery — CHANGES_REQUESTED** [review]

File: packages/core/src/render3/instructions.ts (hunk 1 above)
Comment (marclaval) [reply packages/core/src/render3/instructions.ts]: Merged, PR updated and rebased

**Comment by marclaval** [comment]
Rebased, please have a look

**Comment by mhevery** [comment]
merged as 8315581532

**Pull request closed without merging**